contentcuration/contentcuration/views/admin.py

```diff
@@ -141,6 +142,149 @@ def get_all_users(request):
     return Response(user_serializer.data)
 
 
+class ChannelUserListPagination(PageNumberPagination):
+    page_size = 500
+    page_size_query_param = 'page_size'
+    max_page_size = 500
+
+AND_ = lambda x: 'and_'+x
+OR_ = lambda x: 'or_'+x
+
+@login_required
+@api_view(['GET'])
+@authentication_classes((SessionAuthentication, BasicAuthentication, TokenAuthentication))
+@permission_classes((IsAdminUser,))
+def get_users(request):
+    if not request.user.is_admin:
+        raise SuspiciousOperation("You are not authorized to access this endpoint")
+
+    params = request.GET
+    # return "hey!"
+    paginator = ChannelUserListPagination()
+    user_list = User.objects.prefetch_related('editable_channels').prefetch_related('view_only_channels').distinct()
+    if 'channel_id' in params:
+        channel = Channel.objects.get(pk=params['channel_id'])
+        user_list = user_list.filter(editable_channels__in=[channel])
+    
+    if AND_('is_chef') in params:
+        # import ipdb;ipdb.set_trace()
+        user_list = user_list.annotate(chef_channels=Count('editable_channels__ricecooker_version'))
+        if params.get(AND_('is_chef')) == 'True':
+            user_list = user_list.filter(chef_channels__gt=0)
+        else:
+            user_list = user_list.filter(chef_channels=0)
+
+    query = None
+    for prop in ['is_admin', 'is_active']:
+        if AND_(prop) in params:
+            q = Q(**{prop: True})
+            q = q if params[AND_(prop)] == 'True' else ~q
+            query = q & query if query else q
+        if OR_(prop) in params:
+            q = Q(**{prop: True})
+            q = q if params[OR_(prop)] == 'True' else ~q
+            query = q | query if query else q
+    
+    if query:
+        user_list = user_list.filter(query)
+
+    if params.get('sort_by') in ['email','fist_name','last_name','channels']:
+        if params.get('ordering') == 'desc':
+            ordering = F(params.get('sort_by')).desc()
+        else:
+            ordering = F(params.get('sort_by')).asc()
+
+        user_list = user_list\
+                    .annotate(channels=Count('editable_channels'))\
+                    .order_by(ordering)
+
+    user_list_page = paginator.paginate_queryset(user_list, request)
+    user_serializer = AdminUserListSerializer(user_list_page, many=True)
+
+    return Response(user_serializer.data)
+
+channel_filters = {
+    'all': Q(),
+    'is_live': ~Q(deleted=True),
+    'is_published': Q(published=True),
+    'is_public': Q(public=True),
+    'is_private': ~Q(public=True),
+    'can_edit': Q(can_edit=True),
+    'is_staged': ~Q(staging_tree=None),
+    'is_ricecooker': ~Q(ricecooker_version=None),
+    'is_deleted': Q(deleted=True)
+}
+
+channel_sortable_keys = [
+    'name',
+    'id',
+    'priority', 
+    'users', 
+    # 'items',      # not working yet!
+    # 'modified',       # not working yet!
```

Comment:
Both the `items` and `modified` fields come from the serialization process, and I had trouble converting them into querysets, as they rely on `channel.main_tree.get_descendants()...`.  Anyone know how to do this?

This is something that could possibly be resolved by upgrading to Django 1.11, which supports [subquery expressions](https://docs.djangoproject.com/en/2.0/ref/models/expressions/#subquery-expressions).

Reply:
For the `modified` field, I'm thinking it might actually make sense to change the ContentNode's `save` method to also set the ancestors' `changed` fields to True (instead of [this](https://github.com/learningequality/studio/blob/develop/contentcuration/contentcuration/models.py#L759),  use `self.get_ancestors(include_self=True).update(changed=True)` and I think modified should auto update as well). This would allow you to use `main_tree__modified`

For the `items` field, I don't know if there's a way to neatly put it in the query (see [here](https://stackoverflow.com/questions/47979998)), so you may need to handle it separately